src/main.tsx

```diff
@@ -1,60 +1,103 @@
-import { Navigate, Route, Routes } from "react-router-dom";
 import { useToast } from "@chakra-ui/react";
-import Header from "./components/header/view";
-import Unauthorised from "./features/unauthorised/view";
+import { Route, Routes } from "react-router-dom";
 import Footer from "./components/footer/footer";
+import Header from "./components/header/view";
 import { Selectors as errorSelectors } from "./features/errorHandler/errorHandler";
 import {
   resetNotification,
   selectNotification,
 } from "./features/notifications/notification";
+import Unauthorised from "./features/unauthorised/view";
 
 import { Box } from "@chakra-ui/react";
 
 import SimpleSidebar from "./components/sideBar/sideBar";
 
+import { ReactElement, useEffect } from "react";
+import { useAppDispatch, useAppSelector } from "./app/hooks";
 import { ProblemDetailsModal } from "./componentsCommon/problemDetailsModal/problemDetailsModal";
+import { DetectLoggedInUser } from "./features/authentication/authenticationSlice";
 import JsonPlaceHolderView from "./features/jsonPlaceholderAPI/JsonPlaceHolder";
-import { useAppDispatch, useAppSelector } from "./app/hooks";
-import PlaygroundView from "./features/playground/playgroundView";
-import ConfigTableExampleView from "./features/configTable/configTableExample";
-import MovieTableView from "./features/paginatedTable/moviePage";
-import VideoGamesTableView from "./features/formExample/videoGamesPage";
+import { AuthenticationCallback } from "./lib/authentication/authenticationCallback";
+import { useAuthContext } from "./lib/authentication/authenticationContext";
 import { mainSections } from "./siteMap/mainSections";
-import { useEffect } from "react";
+
+//export const authProvider = new Auth0AuthProvider(authConfig);
+
+function initPath(
+  path: string,
+  element: () => React.ReactNode,
+  authenticatedOnly: boolean = true,
+  userIsLogged: boolean = true
+): ReactElement {
+  const routeElement = element();
+  const finalElement =
+    (authenticatedOnly && userIsLogged) || !authenticatedOnly ? (
+      routeElement
+    ) : (
+      <Unauthorised />
+    );
+  return <Route path={path} element={finalElement} />;
```

Comment:
why is this approach preferrable to always render 

<AuthenticatedOnly><Stuff /></AuthenticatedOnly>

export function AuthenticatedOnly = ({children} : { children: ReactNode }) => {
    const { isLogged } = useAuthContext();
    return isLogged ? {children} : null;
}

Reply:
because if you choose to show the menu item it make sense to create the path that brigns you to the <Unauthorized/> otherwise you'll have a 404 unless to handle also this in the menu.